pinot-tools/src/main/java/org/apache/pinot/tools/perf/QueryRunner.java

```diff
@@ -275,7 +277,7 @@ public static void singleThreadedQueryRunner(PerfBenchmarkDriverConf conf, List<
       for (String query : queries) {
         if (timeout > 0 && System.currentTimeMillis() - startTimeAbsolute > timeout) {
           LOGGER.warn("Timeout of {} sec reached. Aborting", timeout);
-          return;
+          throw new TimeoutException("Timeout of " + timeout + " sec reached. Aborting");
```

Comment:
the intended behavior here may be to simply run as many requests as possible and then stopping. just a thought

Reply:
btw the query runner has a potential inconsistency in its reporting where it may reset some stats every X min during a run

Reply:
> the intended behavior here may be to simply run as many requests as possible and then stopping. just a thought

Chatted with Jackie on this, he said that QueryRunner isn't really used now so we can modify it according to our needs.